packages/fuels-contract/src/contract.rs

```diff
@@ -75,12 +157,12 @@ impl<D> CallResponse<D> {
             .expect("could not retrieve gas used from ScriptResult")
     }
 
-    pub fn new(value: D, receipts: Vec<Receipt>) -> Self {
-        Self {
-            value,
-            gas_used: Self::get_gas_used(&receipts),
-            receipts,
-        }
+    pub fn get_logs(&self) -> Result<Vec<String>, Error> {
```

Comment:
Since this and `get_logs_with_type<T>` are very user-facing, we should have doc-level comments here.